### DISCUSSION TITLE
Unable to run "dev" version of .eslintrc.json

### DISCUSSION THREAD

**Discussion by ghost** [eslint/eslint#16440] [Q&A] [answered]
I have an Angular codebase that has 4 different folders where we are running rules. The root directory has one file we are linting, so technically 5.

~/apps/webtv/.eslintrc.json
~/apps/webtv-e2e/.eslintrc.json
~/libs/metadata/.eslintrc.json
~/libs/recommender/.eslintrc.json
~/.eslintrc.json

Along each ".eslintrc.json" file I also have a ".eslintrc.dev.json", that is currently just a complete copy of the the corresponding file. I would like to change the rules so devs can run those linting rules locally.

I currently call, and am successful, with 

`time TIMING=1 eslint '**/*' --debug`

It complete in ~30 seconds and all the warning/errors are correct as I can tell. 

I have tried the following commands and I get all different issues that takes ~3 minutes to complete:

```
    time TIMING=1 eslint '**/*' --debug --config .eslintrc.dev.json --config libs/metadata/.eslintrc.dev.json --config libs/recommender/.eslintrc.dev.json --config apps/webtv-e2e/.eslintrc.dev.json --config apps/webtv/.eslintrc.dev.json
    time TIMING=1 eslint '**/*' --debug --config .eslintrc.dev.json libs/metadata/.eslintrc.dev.json libs/recommender/.eslintrc.dev.json apps/webtv-e2e/.eslintrc.dev.json apps/webtv/.eslintrc.dev.json
    time TIMING=1 eslint '**/*' --no-eslintrc --debug --config .eslintrc.dev.json --config libs/metadata/.eslintrc.dev.json --config libs/recommender/.eslintrc.dev.json --config apps/webtv-e2e/.eslintrc.dev.json --config apps/webtv/.eslintrc.dev.json
    time TIMING=1 eslint '**/*' --no-eslintrc --debug --config .eslintrc.dev.json libs/metadata/.eslintrc.dev.json libs/recommender/.eslintrc.dev.json apps/webtv-e2e/.eslintrc.dev.json apps/webtv/.eslintrc.dev.json
```

How can I have a "dev" version of my linting rules that can be called in it's place for all 5 files in one command?

**Comment by mdjermanovic** [comment, marked as answer]
Specifying multiple config files is not supported. If you pass multiple `--config` options, only the last one will be used. If you pass multiple files after `--config`, the first be used as the config file while the remaining are treated as files to lint.

`eslint '**/*' --config a.json --config b.json` is the same as `eslint '**/*' --config b.json`
`eslint '**/*' --config a.json b.json` is the same as `eslint '**/*' b.json --config a.json` (lint `'**/*'` and `b.json`, both using config `a.json`)

For your use case, you could try with a single `.eslintrc.json`/`.eslintrc.dev.json` (in the root) that would contain [`"overrides"`](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns) for each of the 4 folders, instead of using nested `.eslintrc.json`/`.eslintrc.dev.json` config files. Future versions of ESLint, with the [new config system](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new), will not support nested config files anyway.

**Reply by ghost** [reply]
what is the best way to override for specific folders, would I do "apps/webtv/**/*.ts"? and do that for each file type specified in my files?

**Reply by mdjermanovic** [reply]
> would I do "apps/webtv/**/*.ts"? and do that for each file type specified in my files?

If you have a common base config for all files in `apps/webtv/` (e.g., some rules that should apply to all file types) then you could have one override for `"apps/webtv/**"`, and then also additional overrides for each file type (e.g., `"apps/webtv/**/*.ts"`) that need additional configuration.
